include/functions_config.inc.php

```diff
@@ -559,6 +559,7 @@ function serendipity_authenticate_author($username = '', $password = '', $is_has
 
         $rows =& serendipity_db_query($query, false, 'assoc');
         if (is_array($rows)) {
+            $is_valid_user = false;
```

Comment:
I believe thats a dual assignment, the lower one can be removed (2 lines down)

Reply:
No, this is needed. `if ($is_valid_user) continue;` will throw an error if the variable is not assigned yet.

Reply:
I did not mean that specific line, but the one after. We have two $is_valid_user = false assignments now.

Reply:
Hm. Wanted to explain that we need both, one for the first check, the second for the variable to be set back if it was true once. But no, if `$is_valid_user` is true once the loop will exit anyway (or rather, continue again and again, which should be a break probably). In this case we can make the continue a break and remove the second assignment.

Reply:
Changed in the new PR